scipy/signal/tests/test_signaltools.py

```diff
@@ -608,35 +762,38 @@ def test_lfiltic_bad_zi(self):
 
 
 class TestLinearFilterFloat32(_TestLinearFilter):
-    dt = np.float32
+    dtype = np.dtype('f')
 
 
 class TestLinearFilterFloat64(_TestLinearFilter):
-    dt = np.float64
+    dtype = np.dtype('d')
 
 
 class TestLinearFilterFloatExtended(_TestLinearFilter):
-    dt = np.longdouble
+    dtype = np.dtype('g')
 
 
 class TestLinearFilterComplex64(_TestLinearFilter):
-    dt = np.complex64
+    dtype = np.dtype('F')
 
 
 class TestLinearFilterComplex128(_TestLinearFilter):
-    dt = np.complex128
-
+    dtype = np.dtype('D')
 
-class TestLinearFilterComplexxxiExtended28(_TestLinearFilter):
-    dt = np.longcomplex
 
+class TestLinearFilterComplexExtended(_TestLinearFilter):
+    dtype = np.dtype('G')
 
 class TestLinearFilterDecimal(_TestLinearFilter):
-    dt = np.dtype(Decimal)
+    dtype = np.dtype('O')
+
+    def type(self, x):
+        return Decimal(str(x))
 
 
 class TestLinearFilterObject(_TestLinearFilter):
-    dt = np.object_
+    dtype = np.dtype('O')
+    type = float
```

Comment:
Ah OK, I see now where it's used. ``type`` is a builtin name, so this just looked a bit off. Never mind.